compiler/noirc_evaluator/src/acir/acir_variable.rs

```diff
@@ -1568,6 +1566,33 @@ impl<F: AcirField, B: BlackBoxFunctionSolver<F>> AcirContext<F, B> {
     }
 }
 
+pub(super) fn power_of_two<F: AcirField>(power: u32) -> F {
+    let full_bytes = power / 8;
+    let extra_bits = power % 8;
+    let final_byte: u8 = match extra_bits % 8 {
+        0 => 0x01,
+        1 => 0x02,
+        2 => 0x04,
+        3 => 0x08,
+        4 => 0x10,
+        5 => 0x20,
+        6 => 0x40,
+        7 => 0x80,
+        _ => unreachable!("We cover the full range of x % 8"),
+    };
+
+    let mut bytes_le: Vec<u8> =
+        std::iter::repeat(0).take(full_bytes as usize).chain(std::iter::once(final_byte)).collect();
+
+    let bytes_be = {
+        // Reverse to get bytes in big endian.
+        bytes_le.reverse();
```

Comment:
Can you generate it in the correct order? Alternatively it's a good use case for `F::from_le_bytes_reduce` added in https://github.com/noir-lang/noir/pull/7258